3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,59 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
-}
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+
+
+async function fetchData(url) {
   // TODO complete this function
+  const response = await fetch(url);
+  const json = await response.json();
+  return json.results;
```

Comment:
It is best to leave a generic, reusable function such as `fetchData` agnostic about the data it fetches. Here, you are making an assumption that the data will have a `.results` property. Because of this, this function is no longer reusable: it will only work if the fetched data is an object with a `.results` property. It is better to let the function that calls `fetchData` access the `.results` property.